applications/DEMApplication/python_scripts/sphere_strategy.py

```diff
@@ -359,15 +359,19 @@ def _UpdateTimeInModelParts(self, time, is_time_to_print = False):
         self._UpdateTimeInOneModelPart(dem_inlet_model_part, time, is_time_to_print)
         self._UpdateTimeInOneModelPart(rigid_face_model_part, time, is_time_to_print)
 
+        self._MoveAllMeshes(time, self.dt)
+
     def _UpdateTimeInOneModelPart(self, model_part, time, is_time_to_print = False):
         model_part.ProcessInfo[TIME] = time
         model_part.ProcessInfo[DELTA_TIME] = self.dt
         model_part.ProcessInfo[TIME_STEPS] += 1
         model_part.ProcessInfo[IS_TIME_TO_PRINT] = is_time_to_print
 
     def FinalizeSolutionStep(self):
-        time = self.spheres_model_part.ProcessInfo[TIME]
```

Comment:
Are you sure we can remove this update of `time`?

Reply:
that variable was being used by the moveAllMeshes() call that is no longer inside FinalizeSolutionStep(). Also, moving that method out of the FinalizeSolutionStep() is what causes the small divergence with the existing reference values (and hence the update of the those values)